Fix charts

alexbruy opened the pull request:
Fixes #585

alexbruy commented:
It works fine here using random rasters
![Peek 2022-04-11 16-27](https://user-images.githubusercontent.com/776954/162876665-c64ee8cb-5487-451a-a99d-f6ee6b5edc11.gif)

I can't check with real data as I get python errors using the latest versions when calculating indicators and accessing selector widgets.

azvoleff commented:
@alexbruy Can you please send any errors you get? There shouldn't be any issues with the latest version (and there haven't been for some time now), so not sure what you are seeing on your end.

alexbruy commented:
Seems 75cd5577ffa57ecdd98a3cbce594fc67a7370002 breaks expression function which was working fine otherwise (except minor issue with type casting) and as described in https://github.com/ConservationInternational/trends.earth/issues/565#issuecomment-1055511989 and below.

azvoleff commented:
Thanks. It didn't work for me before that commit either though. Is it right that you were never able to test and confirm this was working within the actual plugin?

alexbruy commented:
Initial code works for me with the minor type casting fix. Previously it worked even without it, seems something changed after I have updated several Python packages.

azvoleff commented:
Do you mean 36dbe9cd14942c14415b40fa5b31141697df2839? On that commit the charts still don't work for me:

![image](https://user-images.githubusercontent.com/107753/164461349-bf099e46-ec80-4ddb-9eb4-88c1e3768361.png)

timlinux commented:
@alexbruy could you test this against the develop branch. We tested this last week and it seems to work well and will be a better test bed for you to validate the charts are working.

alexbruy commented:
I rebased against `develop` and reverted https://github.com/ConservationInternational/trends.earth/commit/75cd5577ffa57ecdd98a3cbce594fc67a7370002. For me this branch work fine and charts are there. Unfortunately I'm not able to get rasters with the plugin which match requirements confirmed here https://github.com/ConservationInternational/trends.earth/issues/565#issuecomment-1055511989 so I tested with randomly generated files.

azvoleff commented:
Sounds like this PR isn't going to work then. 75cd5577ffa57ecdd98a3cbce594fc67a7370002 was added because as I noted [here](https://github.com/ConservationInternational/trends.earth/issues/585#issuecomment-1080950020) the values for several layers are not 0/1/-1. We need it to be tested on and to work with the data on within the plugin - not just random data.

timlinux commented:
@azvoleff Can you share the list of translation from classes to -1 / 0 / 1 encodings so that @alexbruy can translate the raster values to degraded / unchanged / improved classes.

timlinux commented:
@alexbruy can you please test with a large dataset and if needed introduce some generatlisation logic - it could be a simple approach like reading every nth pixel or something smarter.

azvoleff commented:
As noted on the call, the codes that are charted need to be changed. The codes are:

Land cover:
- Degraded: -1
- Stable: 0
- Improved: 1

Soil organic carbon:
- Degraded: >= -101 and < -10
- Stable: >= -10 and <=10
- Improved: > 10

Land productivity:
- Degraded: 1, 2
- Stable: 3
- Improved: 4, 5

As noted above, https://github.com/ConservationInternational/trends.earth/commit/75cd5577ffa57ecdd98a3cbce594fc67a7370002 was added to address this issue

azvoleff commented:
The charts still don't show. Have you tested and confirmed this is working for you on real data from within the plugin?

alexbruy commented:
> The charts still don't show. Have you tested and confirmed this is working for you on real data from within the plugin?

Seems you have tested old code. Everything is working now.

azvoleff commented:
No, I don't think so. I was on b6aaf43, and now just tested.

Have you tested and confirmed this is working for you on real data produced from within the plugin?

azvoleff commented:
I still can't get the charts to show for me. I've tested on QGIS 3.24 on both Windows and Linux, using a number of different countries (including the Botswana example I mentioned before). None of them work for me. It is possible it is something in my setup, but if so odd it would occur on across two operating systems. @gkahiu - can you try on your machine and see if the charts work for you?

gkahiu commented:
Not sure whether am doing it correctly but am not getting the charts. See below:

![image](https://user-images.githubusercontent.com/3840267/167713964-fe858aa4-a96c-4d22-ac34-5aaa12965c28.png)

azvoleff commented:
Thanks. That's what I see as well. @alexbruy - is it possible there is something on your setup that differs? What QGIS version have you been testing on?

alexbruy commented:
![зображення](https://user-images.githubusercontent.com/776954/168079317-65da5c3d-0fdd-4cec-8f0d-2c2b4d462dac.png)

azvoleff commented:
@alexbruy - is it possible there is something on your setup that differs? What QGIS version have you been testing on?

azvoleff commented:
@timlinux have you been able to get this working on your setup? As I noted above I've tested now on Win and LInux and the charts don't work on either. Same I believe is the case for John. It seems there must be something different about Alex's install (possibly a development QGIS version?). Not sure how best to track this down.

alexbruy commented:
> is it possible there is something on your setup that differs?

I can't tell, as I know nothing about your settings and setup. I've asked (at the meetings and [here](https://github.com/ConservationInternational/trends.earth/pull/615#issuecomment-1118534945)) about it but get no information yet. May I as once again ask what settings you use? What are the exact steps you follow? Are these steps in line with the workflow outlined in the other tickets?

> What QGIS version have you been testing on?

For me it works on all supported versions both on Linux and Windows.

azvoleff commented:
Thanks @alexbruy. See my prior responses below (copied from above). Do you need more than this? I'm not sure what you mean by "settings" beyond what I have provided.

> I still can't get the charts to show for me. I've tested on QGIS 3.24 on both Windows and Linux, using a number of different countries (including the Botswana example I mentioned before), and on both the latest you linked above ([2d502e6](https://github.com/ConservationInternational/trends.earth/commit/2d502e6db873c3cd74e423f1180be8bf28f6ed2a)) as well as on this PR updated to match the current head ([82f81dd](https://github.com/ConservationInternational/trends.earth/commit/82f81ddb436e7a553f51b1252a1e085a3eb9fe0a)). None of them work for me.
> 
>It is possible it is something in my setup, but if so odd it would occur on across two operating systems.

alexbruy commented:
I'm not sure either what you mean by " something on your setup that differs". Can you share any information on following

> What are the exact steps you follow? Are these steps in line with the workflow outlined in other tickets?

azvoleff commented:
I follow the process you [outlined in the docs](https://trendsearth.readthedocs.io/en/latest/for_users/training/specialareas.html):

https://user-images.githubusercontent.com/107753/168757515-c3fffa2e-a0a9-4c7f-a2ff-f0606b6c77de.mp4

alexbruy commented:
I followed the steps from your screencast with the same region.
![Peek 2022-05-18 09-15](https://user-images.githubusercontent.com/776954/168970326-9904cec0-787c-4a5a-931b-52cf74af4f40.gif)

alexbruy commented:
Completely clean environment
![Peek 2022-05-18 09-58](https://user-images.githubusercontent.com/776954/168977264-1b964d6c-c1b2-4d24-9199-3a9219b780a0.gif)

azvoleff commented:
@alexbruy those both look to be the development release of QGIS. Does it work for you on 3.22?

alexbruy commented:
See above

> For me it works on all supported versions both on Linux and Windows.

alexbruy commented:
![Peek 2022-05-18 16-32](https://user-images.githubusercontent.com/776954/169194612-21ac6306-e93b-44b5-9b9c-7aff7d040ccb.gif)
